add missing verify_fun mfa translation

danj3 opened the pull request:
inet_tls_dist is missing a configuration translation for verify_fun from an mfa tuple to a closure that can be called during ssl_handshake.

@IngelaAndin This is the missing verify_fun translation I spoke of, applied to master

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/erlang/otp?pullRequest=9692) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/erlang/otp?pullRequest=9692) before we can accept your contribution.<br/><hr/>**Dan Janowski** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/erlang/otp?pullRequest=9692) it.</sub>

danj3 commented:
Ah I forgot something. This change is to support -ssl_dist_optfile that looks like:
```
[{server,[{fail_if_no_peer_cert,true},
          {certfile,"erldist_cert.pem"},
          {keyfile,"erldist_privkey.pem"},
          {cacertfile,"erldist_cert.pem"},
          {verify,verify_peer},
          {verify_fun,{localca,verify,"erldist_cert.pem"}}]},
 {client,[{certfile,"erldist_cert.pem"},
          {keyfile,"erldist_privkey.pem"},
          {cacertfile,"erldist_cert.pem"},
          {verify,verify_peer},
          {verify_fun,{localca,verify,"erldist_cert.pem"}}]}].
```

These quasi-mfa verify_fun tuples are turned into the closure functions by this patch. The existing code was ignoring/passing over the verify_fun of this form. This form is based on what I got out of the [Erlang Distribution over TLS](https://www.erlang.org/doc/apps/ssl/ssl_distribution.html#specifying-tls-options)

IngelaAndin commented:
I do not think you should base your new code on how legacy option handling was done. The latest version of 
the documentation that you are referring to, of erlang distribution, clearly states:
"The following section describes TLS Option handling prior to OTP 20.2 and can only handle a small subset of the actual available options. It is here only for the sake of backwards compatibility ." And I would claim that it is an error in the documentation because verify_fun has always been` {verify_fun, {fun(), UserState::term()}}`

RaimoNiskanen commented:
I just pointed out that the documentation is not incorrect.

I see no point in adding support for this legacy format to `-ssl_dist_optfile`, and did not argue for that.

danj3 commented:
I agree that supporting the legacy verify_fun option in the optfile is pointless. I added an ssl doc section and reverted the code change in the #9691 the pr to make very clear the specific use of verify_fun in the optfile. While the example is implied by the existing documentation, a concrete example of this kind would have saved me considerable time. I can update this PR as well at your request, as well as any content changes to the doc.

danj3 commented:
The intended change appears lost during the rebase to otp-28. I've re-rebased with the intended change, duplicate of the doc add to otp-27.

RaimoNiskanen commented:
Closing this since the twin PR #9691 should be able to merge forward all the way.

Sorry for the confusion and causing redundant work.

Thank you for your contribution!
